tests/baselines/reference/APISample_WatchWithOwnWatchHost.js

```diff
@@ -36,8 +36,8 @@ function watchMain() {
         readDirectory: ts.sys.readDirectory,
         realpath: ts.sys.realpath,
 
-        watchFile: ts.sys.watchFile!,
-        watchDirectory: ts.sys.watchDirectory!,
+        watchFile: ts.sys.watchFile! as any,
```

Comment:
what?

Reply:
ts.sys.watchFile is a function with a final parameter of type WatchOptions, which has a property of type WatchFileKind, a numeric enum. CompilerHost's watchFile, the target signature, has a final parameter of type CompilerOptions, which does *not* have a matching property, but *does* have an index signature that includes number. But number isn't assignable to an enum.

This is technically sound, because people could call `watchFile` with an options bug containing any number at all (or a lot of other types), but `ts.sys.watchFile` can only handle WatchFileKind on certain of its properties.

However, I don't know how likely this error is to happen in practise. We haven't had trouble as far as we know.